lib/hammer_cli/defaults.rb

```diff
@@ -0,0 +1,143 @@
+
+      def execute
+        begin
+          if option_plugin_name.nil? && option_param_val.nil? || !option_plugin_name.nil? && !option_param_val.nil?
+            bad_input
+          else
+            if option_plugin_name
+              namespace = ("HammerCLI" + option_plugin_name.split('_').collect!{ |w| w.downcase.capitalize }.join + "::Defaults").to_sym
+              raise NameError unless HammerCLI::Defaults.providers.keys.include?(namespace)
+              raise NotImplementedError unless HammerCLI::Defaults.providers[namespace].support? option_param_name
+            end
+            Defaults.add_defaults_to_conf({option_param_name => option_param_val}, namespace)
+            added_default_message(option_param_name.to_s, option_param_val)
+          end
+          rescue NameError => e
+          plugin_prob_message
+          rescue StandardError => e
+          file_not_found_message
+          rescue NotImplementedError => e
+          defaults_not_supproted_by_plugin
+          HammerCLI::EX_CONFIG
+        end
+        HammerCLI::EX_OK
+      end
+    end
+
+    def added_default_message(key, value)
+        print_message(_("Added %{key_val} default-option with value that will be generated from the server.") % {:key_val => key.to_s} ) if value.nil?
+        print_message(_("Added %{key_val} default-option with value %{val_val}.") % {:key_val => key.to_s, :val_val=> value.to_s} ) unless value.nil?
+    end
+
+    def plugin_prob_message
+      print_message(_("Couldn't reach the plugin defaults class, eiter the plugin doesn't have defaults class or the plugin misspelled.(example, for Hammer_CLI_Foreman --plugin-name foreman)"))
+    end
+
+    def defaults_not_supproted_by_plugin
+      print_message(_("The param name is not supported by plugin"))
+    end
+
+
+    def bad_input
+      print_message(_("You must specify value or a plugin name, cant specify both."))
+    end
+
+    def self.file_cant_be_created
+      print_message(_("Couldn't add file to ~/.hammer/cli.modules.d please create the path before defaults will be enabled."))
+    end
+
+    def list_all_defaults_message
+      unless HammerCLI::Settings.settings[:defaults].nil?
+        print_message("Hammer defaults list")
+        print_message("--------------------")
+        HammerCLI::Settings.settings[:defaults].each do |key,val|
+          if HammerCLI::Settings.settings[:defaults][key][:from_server]
+            print_message(key.to_s + " : " +  _("(provided by %{plugin})") % {:plugin => HammerCLI::Settings.settings[:defaults][key][:provider].to_s.split(':').first.gsub("HammerCLI", '')})
+          else
+            print_message(key.to_s + " : " + HammerCLI::Settings.settings[:defaults][key][:value].to_s)
+          end
+        end
+      else
+        print_message(_("No defaults file was found"))
+      end
+    end
+
+    autoload_subcommands
+  end
+
+
+  class Defaults
+    def self.register_provider(provider)
```

Comment:
Use separate variables for the provider name and the provider, please. It will increase readability of the code.